library/src/main/java/org/apache/apex/malhar/lib/window/Window.java

```diff
@@ -18,31 +18,42 @@
  */
 package org.apache.apex.malhar.lib.window;
 
-import java.util.Comparator;
-
 import org.apache.hadoop.classification.InterfaceStability;
 
 /**
- * This interface describes the individual window.
+ * This interface describes the classes that represent individual windows.
+ *
+ * @param <WINDOW> window type the object of this class can call compareTo
  *
  * @since 3.5.0
  */
 @InterfaceStability.Evolving
-public interface Window
+public interface Window<WINDOW extends Comparable<WINDOW>> extends Comparable<WINDOW>
 {
   long getBeginTimestamp();
 
   long getDurationMillis();
 
   /**
-   * Global window means there is only one window, or no window depending on how you look at it.
+   * Global window means there is only one window that spans the entire life time of the application
    */
-  class GlobalWindow implements Window
+  class GlobalWindow implements Window<GlobalWindow>
   {
+
+    /**
+     * The singleton global window
+     */
+    private static GlobalWindow GLOBAL_WINDOW = new GlobalWindow();
+
     private GlobalWindow()
     {
     }
 
+    public static GlobalWindow getInstance()
```

Comment:
I'm also curious about this change and to me. And also the equals method can use == instead of instanceof which I believe would be slightly faster